src/azul/transformer.py

```diff
@@ -243,19 +254,34 @@ def to_index(self, field_types, bulk=False, update=False) -> dict:
             **({} if delete else {'_source' if bulk else 'body': self.translate_fields(self.to_source(), field_types)}),
             '_id' if bulk else 'id': self.document_id
         }
-        if self.version_type is None:
-            # TODO: FIXME: (jesse) why only for bulk???
-            if False and bulk:
+        if self.version_type is VersionType.none:
+            assert self.version is None
+            if bulk:
                 result['_op_type'] = 'delete' if delete else 'index'
-            op_key = '_op_type' if bulk else 'op_type'
```

Comment:
In fact, I would probably remove all of the `_key if bulk else key` redundancies in this method using a local function like this:

```
def result_key(keyname):
  bulk_keys = {
     'index': '_index',
     'doc_type': '_type',
     'body': '_source',
     'id': '_id',
     'op_type': '_op_type',
     'version_type': '_version_type', 
     'version': '_version'
  }
  return bulk_keys[keyname] if bulk else keyname
```

And then:

```
result = { 
  result_key('index'): self.document_index,
  result_key('doc_type'): self.type
  ...
}
op_key = result_key('op_type')
if self.version_type is VersionType.none:
  assert self.version is None
  if bulk:
    result[op_key] = 'delete' if delete else 'index'
  ...
    result[result_key('version_type')] = 'internal'
    ...
```
This isn't as important for the keys that are only used in one place though.

Reply:
Actually, if it were 100% up to me, I'd create separate methods for the bulk and non-bulk case, or create classes to streamline handling each case (replace these nested `if` with polymorphic method calls). But I'm not convinced that's consistent with design decision precedent for Azul so I won't request it. If you think it's a good idea though you should do it since this method is the one part of the code where these changes _increased_ complexity.

Reply:
You inspired me to try a completely different approach. I tried first splitting on `if bulk:` I think the makes the `result_key()` method redundant (though I liked the idea). I think this is more clear than before, but not as extreme as making separate classes.